azkaban-common/src/main/java/azkaban/jobtype/JobTypeManager.java

```diff
@@ -433,14 +437,32 @@ public JobParams createJobParams(final String jobId, Props jobProps, final Logge
           jobProps.put(key, clusterSpecificProps.get(key));
         }
       }
+
+      // Override any plugin load props if specified.
+      Props pluginLoadPropsCopy = pluginLoadProps;
+      if (pluginLoadOverrideProps != null) {
+        // Make a clone of pluginLoadProps
+        pluginLoadPropsCopy = Props.clone(pluginLoadProps);
+        final String[] propsList = pluginLoadOverrideProps.split(",");
+        for (final String prop : propsList) {
+          final String value = clusterSpecificProps.getString(prop, null);
+          if (value == null) {
+            // The property must be present in cluster specific props
+            throw new JobExecutionException(String.format("Required cluster property %s "
```

Comment:
If property isn't present then we should ignore. Shouldn't we?

Reply:
The set of properties must be there as we should list only the mandatory ones.

Reply:
I think we should ignore rather than failing. Log error and go ahead.

Reply:
Ack